src/python/pants/util/resources.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+
+import importlib
+from itertools import chain
+
+import importlib_resources as resources
+
+
+def read_resource(package_or_module: str, resource: str) -> bytes:
+    """Reads a resource file from within the Pants package itself.
+
+    This helper function is designed for compatibility with `pkgutil.get_data()` wherever possible,
+    but also allows compability with PEP302 pluggable importers such as included with PyOxidizer.
+    This requires that resources are loaded from a valid Python package (i.e. must have an
```

Comment:
Per PEP 420, a directory without an `__init__.py` is a valid Python package (in fact it is a namespace package). The true restriction here is that we require an `__init__.py` because of a limitation of importlib_resources, but it is an overstatement to say that the lack of one makes a package not valid.